### DISCUSSION TITLE
Bindings for REST API

### DISCUSSION THREAD

**Discussion by datadumppro** [Budibase/budibase#8142] [Help] [answered]
Hello! 

This probably isn't possible, but is there a way to incorporate bindings into the REST API function without the {{ start }} escaping in the URL? It is causing the API call to fail as it is an invalid date format

I input the parameters in the screenshot below but the URL keeps reverting to: https://api.aeso.ca/report/v1.1/price/poolPrice?startDate=%7B%7BBinding.start%7D%7D%20&endDate=%7B%7BBinding.end%7D%7D%20

![Bindings](https://user-images.githubusercontent.com/58442501/194095971-455d7283-8842-4712-bb7c-47cad325bf86.JPG)
![Parameters](https://user-images.githubusercontent.com/58442501/194095973-821e018a-2acd-4009-93be-a19e165a6a56.JPG)
![Results](https://user-images.githubusercontent.com/58442501/194097265-4bcd43f5-2b10-483c-b181-c99c5c49fead.JPG)

**Comment by datadumppro** [comment]
I've already incorporated the API Key so that is no issue

**Reply by melohagan** [reply]
Hey @datadumppro 

I'm not able to reproduce this. Are you on the latest version of Budibase?

![Screenshot 2022-10-10 at 10 34 53](https://user-images.githubusercontent.com/101575380/194837204-47980cbe-08d9-460c-8459-b017e1da8fce.png)

![Screenshot 2022-10-10 at 10 35 13](https://user-images.githubusercontent.com/101575380/194837255-8d1320c4-a667-45e4-aa4f-93ebf9d6c550.png)

**Reply by datadumppro** [reply]
When we updated, this fixed the bug for us - although it does still happen from time to time! The only thing we do to fix it thus far is log out/log back in over the course of a couple of days and it eventually snags and properly displays the bindings in the URL.

**Comment by dannyshin76** [comment]
Same thing is happening to me, the URL keeps being unnecessarily encoded everytime I try something:
![image](https://user-images.githubusercontent.com/93232582/199930231-a741bafe-7a90-49e9-bd61-a77409e6c14f.png)

Budibase version: 1.3.15

Could you help?

**Reply by melohagan** [reply]
Are you able to upgrade to v2.1 ?

**Reply by dannyshin76** [reply]
So is it a bug? Whew... I am using a local installation. Can I upgrade with the free plan?

**Reply by melohagan** [reply]
Yeah of course! Using the budi CLI you would do `budi hosting --update`

https://docs.budibase.com/docs/budibase-cli-setup

Otherwise with docker you should be able to do `docker pull budibase/budibase:latest`

**Reply by dannyshin76** [reply]
I tried to migrate to v2.1 but came up with another problem, could you help out? 🙏
https://github.com/Budibase/budibase/discussions/8521

**Reply by andyshinn** [reply]
Did you every figure out the re-encoded params every time? Is there a bug for this? I am on 2.1.24 trying it out and running into this issue.

**Reply by dannyshin76** [reply]
I am also on 2.1.24. Although I still have an issue of recurring "Binding" like this whenever I try to edit:
![image](https://user-images.githubusercontent.com/93232582/201797846-48d7b048-561a-428a-914f-0ab448e339b8.png)

but it is treatable. No re-encoding anyway.

**Reply by datadumppro** [reply]
Same thing happens with us @dannyshin76 but after manually editing it out that solves it on our end.

**Comment by melohagan** [comment, marked as answer]
Looks to be related to this bug: https://github.com/Budibase/budibase/issues/9583
